sycl/include/sycl/device.hpp

```diff
@@ -214,8 +217,37 @@ class __SYCL_EXPORT device : public detail::OwnerLessBase<device> {
   /// type associated with the param parameter.
   ///
   /// \return device info of type described in Table 4.20.
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  template <typename Param>
+  typename detail::is_device_info_desc<Param>::return_type get_info() const {
+    // For C++11_ABI compatibility, we handle these string Param types
+    // separately.
+    if constexpr (std::is_same_v<Param, info::device::name> ||
+                  std::is_same_v<Param, info::device::vendor> ||
+                  std::is_same_v<Param, info::device::driver_version> ||
+                  std::is_same_v<Param, info::device::version> ||
+                  std::is_same_v<Param, info::device::profile> ||
+                  std::is_same_v<Param, info::device::opencl_c_version> ||
+                  std::is_same_v<Param, info::device::ext_intel_pci_address>) {
+      detail::string_view PropertyName(typeid(Param).name());
+      detail::string Info = get_device_info(PropertyName);
+      return Info.c_str();
+    } else if constexpr (std::is_same_v<Param, info::device::extensions>) {
+      // return value is std::vector<std::string>
```

Comment:
That means that instead of explicit properties at line 225 we could do

```
if constexpr (std::is_same_v<std::string, detail::is_device_info_desc<Param>::return_type>)
```

because what we really care about is return type only. Same for `std::vector<std::string>`.

Reply:
No, this is the case where the template paramter `Param `itself is a std::string, and it has to be converted to `detail::string_view` to be passed to libsycl.

Reply:
That line is checking if the template parameter `Param `and `info:;device::extensio`n are the same string value. It is not checking if the return value is a string.
Please note that we cannot pass a `std::string` from the header file to the libsycl side.
We need to convert all `std::string` instances into `detail::string_view` instances.